processing/src/main/java/io/druid/query/topn/types/StringTopNColumnSelectorStrategy.java

```diff
@@ -46,6 +46,10 @@ public int getCardinality(DimensionSelector selector)
     // Optimization possibly requires a reverse lookup from value to ID, which is
     // not possible when applying an extraction function
 
+    if (params.getCardinality() < 0) {
```

Comment:
removed the check, no longer needed with separate code path for "cardinality unknown" case